packages/system/Accounts/plugin/accounts/src/helpers.rs

```diff
@@ -36,3 +36,44 @@ pub fn assert_valid_account(account: &str) {
         AccountsPlugin::get_account(account.to_string()).expect("Get account failed");
     assert!(account_details.is_some(), "Invalid account name");
 }
+
+use psibase::AccountNumber;
+use rand::prelude::*;
+
+pub fn generate_account() -> Result<AccountNumber, CommonTypes::Error> {
+    let mut rng = rand::rng();
+
+    let max_tries = 1000;
+    let mut tries = 0;
+
+    loop {
+        if tries >= max_tries {
+            return Err(MaxGenerationAttemptsExceeded().into());
+        }
+        tries += 1;
+
+        let mut account = String::new();
+
+        // First character: lowercase a-z
+        let first_chars: &[char] = &(b'a'..=b'z').map(|b| b as char).collect::<Vec<_>>();
```

Comment:
You shouldn't be building `first_chars` and `allowed_chars` repeatedly on each loop (in fact these could even be created statically)